File: src_original/extras-kconfig/libkdesktopfileaction.cpp
#include <KDesktopFileAction>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kdesktopfileaction.h>
#include "libkdesktopfileaction.h"
#include "libkdesktopfileaction.hxx"

KDesktopFileAction* KDesktopFileAction_new() {
    return new KDesktopFileAction();
}

KDesktopFileAction* KDesktopFileAction_new2(const libqt_string name, const libqt_string text, const libqt_string icon, const libqt_string exec, const libqt_string desktopFilePath) {
    QString name_QString = QString::fromUtf8(name.data, name.len);
    QString text_QString = QString::fromUtf8(text.data, text.len);
    QString icon_QString = QString::fromUtf8(icon.data, icon.len);
    QString exec_QString = QString::fromUtf8(exec.data, exec.len);
    QString desktopFilePath_QString = QString::fromUtf8(desktopFilePath.data, desktopFilePath.len);
    return new KDesktopFileAction(name_QString, text_QString, icon_QString, exec_QString, desktopFilePath_QString);
}

KDesktopFileAction* KDesktopFileAction_new3(const KDesktopFileAction* other) {
    return new KDesktopFileAction(*other);
}

void KDesktopFileAction_OperatorAssign(KDesktopFileAction* self, const KDesktopFileAction* other) {
    self->operator=(*other);
}

libqt_string KDesktopFileAction_ActionsKey(const KDesktopFileAction* self) {
    QString _ret = self->actionsKey();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KDesktopFileAction_DesktopFilePath(const KDesktopFileAction* self) {
    QString _ret = self->desktopFilePath();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KDesktopFileAction_Name(const KDesktopFileAction* self) {
    QString _ret = self->name();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KDesktopFileAction_Icon(const KDesktopFileAction* self) {
    QString _ret = self->icon();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KDesktopFileAction_Exec(const KDesktopFileAction* self) {
    QString _ret = self->exec();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

bool KDesktopFileAction_IsSeparator(const KDesktopFileAction* self) {
    return self->isSeparator();
}

void KDesktopFileAction_Delete(KDesktopFileAction* self) {
    delete self;
}

File: src_original/extras-kconfig/libkdesktopfileaction.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/kdesktopfileaction.html
pub const kdesktopfileaction = struct {
    /// New constructs a new KDesktopFileAction object.
    ///
    ///
    pub fn New() QtC.KDesktopFileAction {
        return qtc.KDesktopFileAction_new();
    }

    /// New2 constructs a new KDesktopFileAction object.
    ///
    /// ``` name: []const u8, text: []const u8, icon: []const u8, exec: []const u8, desktopFilePath: []const u8 ```
    pub fn New2(name: []const u8, text: []const u8, icon: []const u8, exec: []const u8, desktopFilePath: []const u8) QtC.KDesktopFileAction {
        const name_str = qtc.libqt_string{
            .len = name.len,
            .data = name.ptr,
        };
        const text_str = qtc.libqt_string{
            .len = text.len,
            .data = text.ptr,
        };
        const icon_str = qtc.libqt_string{
            .len = icon.len,
            .data = icon.ptr,
        };
        const exec_str = qtc.libqt_string{
            .len = exec.len,
            .data = exec.ptr,
        };
        const desktopFilePath_str = qtc.libqt_string{
            .len = desktopFilePath.len,
            .data = desktopFilePath.ptr,
        };

        return qtc.KDesktopFileAction_new2(name_str, text_str, icon_str, exec_str, desktopFilePath_str);
    }

    /// New3 constructs a new KDesktopFileAction object.
    ///
    /// ``` other: QtC.KDesktopFileAction ```
    pub fn New3(other: ?*anyopaque) QtC.KDesktopFileAction {
        return qtc.KDesktopFileAction_new3(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#operator-eq)
    ///
    /// ``` self: QtC.KDesktopFileAction, other: QtC.KDesktopFileAction ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.KDesktopFileAction_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#actionsKey)
    ///
    /// ``` self: QtC.KDesktopFileAction, allocator: std.mem.Allocator ```
    pub fn ActionsKey(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KDesktopFileAction_ActionsKey(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kdesktopfileaction.ActionsKey: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#desktopFilePath)
    ///
    /// ``` self: QtC.KDesktopFileAction, allocator: std.mem.Allocator ```
    pub fn DesktopFilePath(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KDesktopFileAction_DesktopFilePath(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kdesktopfileaction.DesktopFilePath: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#name)
    ///
    /// ``` self: QtC.KDesktopFileAction, allocator: std.mem.Allocator ```
    pub fn Name(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KDesktopFileAction_Name(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kdesktopfileaction.Name: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#icon)
    ///
    /// ``` self: QtC.KDesktopFileAction, allocator: std.mem.Allocator ```
    pub fn Icon(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KDesktopFileAction_Icon(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kdesktopfileaction.Icon: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#exec)
    ///
    /// ``` self: QtC.KDesktopFileAction, allocator: std.mem.Allocator ```
    pub fn Exec(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.KDesktopFileAction_Exec(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("kdesktopfileaction.Exec: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#isSeparator)
    ///
    /// ``` self: QtC.KDesktopFileAction ```
    pub fn IsSeparator(self: ?*anyopaque) bool {
        return qtc.KDesktopFileAction_IsSeparator(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kdesktopfileaction.html#dtor.KDesktopFileAction)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KDesktopFileAction ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KDesktopFileAction_Delete(@ptrCast(self));
    }
};
